specs/reservations_spec.rb

```diff
@@ -0,0 +1,200 @@
+      @hotel_res.make_booking(1, (date1 + 1), (date1 + 2)).must_be_kind_of Hotel::Booking
+    end
+  end
+
+  describe 'make a block' do
+    before do
+      @block_booking = @hotel_res.make_booking(5, @start_date, @end_date, block: true)
+    end
+    it "can create a block reservation (and it doesn't mess anything else up.)" do
+      @block_booking.must_be_kind_of Hotel::Booking
+      @block_booking.must_be_kind_of Hotel::Block
+    end
+
+    it 'has 5 rooms, but none are reserved' do
+      @block_booking.rooms.length.must_equal 5
+      @block_booking.rooms.each do |room|
+        room.reserved.must_equal false
+      end
+    end
+
+    it 'can only have 5 rooms per block' do
+      proc {@hotel_res.make_booking(6, @start_date, @end_date, block: true)}.must_raise Hotel::RoomQuantityError
+
+    end
+
+    it 'can reserve a room within a block' do
+      @hotel_res.reserve_block_room(@block_booking, 1)
+      @block_booking.rooms.length.must_equal 5
+      @block_booking.rooms[0].reserved.must_equal true
+      @block_booking.rooms[4].reserved.must_equal false
+    end
+
+    it 'reserving a room creates a new reservation, linked to the block' do
+      room_reserve =   @hotel_res.reserve_block_room(@block_booking, 1)
+
+      room_reserve.id.wont_equal @block_booking.id
+
+      @hotel_res.all_reservations.length.must_equal room_reserve.id
+
+      room_reserve.block_id.must_equal @block_booking.block_id
+
+      room_reserve2 =   @hotel_res.reserve_block_room(@block_booking, 1)
+
+      @hotel_res.all_reservations.length.must_equal room_reserve2.id
+
+    end
+
+    it 'reserving rooms within only works on blocks' do
+      proc {  @hotel_res.reserve_block_room(@booking2, 1)}.must_raise NoMethodError
+    end
+
+    it 'reserved room has the same date_range as block' do
+      room_reserve =   @hotel_res.reserve_block_room(@block_booking, 1)
+      @block_booking.date_range.must_equal room_reserve.date_range
+    end
+
+
+    it 'can return a list of availbe rooms within the block' do
+      @hotel_res.reserve_block_room(@block_booking, 1)
+      @block_booking.rooms_available.length.must_equal 4
+      @block_booking.rooms_available[0].must_be_kind_of Hotel::Room
+    end
+
+    it "can't reserve more rooms than the block has left." do
+      @hotel_res.reserve_block_room(@block_booking, 4)
+      proc {  @hotel_res.reserve_block_room(@block_booking, 2)}.must_raise Hotel::RoomQuantityError
+
+    end
+
+    it 'blocks recieve a 20% discount' do
+      @block_booking.total_cost.must_equal 800
+    end
+
+  end
+
+
+  describe 'check_reserved' do
+    it 'returns an array of rooms' do
+      @hotel_res.check_reserved(@start_date, @end_date).must_be_kind_of Array
+      @hotel_res.check_reserved(@start_date, @end_date)[0].must_be_kind_of Hotel::Room
+    end
+
+    it 'returns an empty/full array if nothing/everything is reserved on that day' do
+      date1 = Date.new(2017,10,1)
+      date2 = Date.new(2017,10,2)
```

Comment:
I like this test case! I'd also like to see a test for how reservations that don't overlap the date range affect the list (they shouldn't).

You could also check the boundaries of what counts as an overlap. What about a reservation that ends on the checkin date, or starts on the checkout date?